Make dropdown caret to right like select dropdown

anshulguleria opened the pull request:
Sticks dropdown caret to right side like in default select dropdown of
browser. Usually in cases where data-toggle button is bigger size, caret
was in center thus was not looking good.

anshulguleria committed:
Make dropdown caret to right like select dropdown

Sticks dropdown caret to right side like in default select dropdown of
broweser. Usually in cases where data-toggle button is bigger size caret
was in center thus was not looking good.

anshulguleria commented on scss/_dropdown.scss:
```diff
@@ -5,11 +5,20 @@
     display: inline-block;
     width: 0;
     height: 0;
+    right: 0.5em;
```
Right align carret at vertical center like in browser default select dropdown

hnrch02 commented on scss/_dropdown.scss:
```diff
@@ -5,11 +5,20 @@
+    -moz-transform: translateY(-50%);
+    -webkit-transform: translateY(-50%);
+    -o-transform: translateY(-50%);
+    -ms-transform: translateY(-50%);
```
You don't need all of the prefixes here, we run [autoprefixer](https://github.com/postcss/autoprefixer).

anshulguleria committed:
Removes prefixes since we have autoprefixer

As pointed out by
[hnrch02](https://github.com/twbs/bootstrap/pull/17116#discussion_r37519039) we use
[autoprefixer](https://github.com/postcss/autoprefixer) so removed the
prefixes.

mdo commented:
JS Bin of the problem this addresses?

anshulguleria commented:
https://jsbin.com/yadomi

I demonstrated with bootstrap3 because alpha4 link is not available but effect is same in both.

mdo commented:
Ah, we won't be doing this. Thanks though!
